syntax = "proto3";
package view;




import "topology_sdk/model/topology/node.proto";
import "topology_sdk/model/topology/link.proto";
import "topology_sdk/model/topology/area.proto";
import "topology_sdk/model/topology/note.proto";

/*
UpdateView请求
*/
message UpdateViewRequest {
message Diff {
    /*
    新增节点
    */
    repeated topology.Node addNodes = 1 ;
    /*
    已删除节点
    */
    repeated topology.Node removeNodes = 2 ;
    /*
    新增线
    */
    repeated topology.Link addLinks = 3 ;
    /*
    已删除线
    */
    repeated topology.Link removeLinks = 4 ;
}
        
    
    /*
    ID
    */
    string id = 1 ;
    /*
    名称
    */
    string name = 2 ;
    /*
    允许可读用户
    */
    repeated string readAuthorizers = 3 ;
    /*
    允许可写用户
    */
    repeated string writeAuthorizers = 4 ;
    /*
    版本
    */
    string version = 5 ;
    /*
    根节点
    */
    topology.Node rootNode = 6 ;
    /*
    节点
    */
    repeated topology.Node nodes = 7 ;
    /*
    线
    */
    repeated topology.Link links = 8 ;
    /*
    区域
    */
    repeated topology.Area areas = 9 ;
    /*
    标注
    */
    repeated topology.Note notes = 10 ;
    /*
    差异
    */
    Diff diff = 11 ;
}

/*
UpdateView返回
*/
message UpdateViewResponse {
    /*
    ID
    */
    string id = 1 ;
}

/*
UpdateViewApi返回
*/
message UpdateViewResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    view.UpdateViewResponse data = 4 ;
}

